src/Generators/EntityGenerator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of Factory Muffin.
+ *
+ * (c) Graham Campbell <graham@alt-three.com>
+ * (c) Scott Robertson <scottymeuk@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\FactoryMuffin\Generators;
+
+use League\FactoryMuffin\FactoryMuffin;
+
+/**
+ * This is the entity generator class.
+ *
+ * The entity generator can be useful for setting up relationships between
+ * models. The entity generator will create and return the generated model.
+ *
+ * @author Graham Campbell <graham@alt-three.com>
+ * @author Scott Robertson <scottymeuk@gmail.com>
+ * @author Michael Bodnarchuk <davert@codeception.com>
+ */
+class EntityGenerator implements GeneratorInterface
+{
+    /**
+     * Definition prefix to match this generator.
+     */
+    const PREFIX = 'entity|';
+    /**
+     * The kind of attribute that will be generated.
+     *
+     * @var string
+     */
+    private $kind;
+
+    /**
+     * The model instance.
+     *
+     * @var object
+     */
+    private $model;
+
+    /**
+     * The factory muffin instance.
+     *
+     * @var \League\FactoryMuffin\FactoryMuffin
+     */
+    private $factoryMuffin;
+
+    /**
+     * Create a new factory generator instance.
+     *
+     * @param string                              $kind          The kind of attribute.
+     * @param object                              $model         The model instance.
+     * @param \League\FactoryMuffin\FactoryMuffin $factoryMuffin The factory muffin instance.
+     */
```

Comment:
strange, PHPStorm marks `@return void` in constructor as error and updates it
I don't feel it is necessary